### DISCUSSION TITLE
Reader Hardware: µHAN mosquito

### DISCUSSION THREAD

**Discussion by dbeinder** [UtilitechAS/amsreader-firmware#480] [Show and tell]
I've been working on a tiny, MBUS-powered board: https://github.com/dbeinder/han-mosquito
  - Power & flashing via USB-C
  - Power & meter signal via M-BUS (RJ11)
    - Input current limited to 40mA, this is probably only compatible with Kaifa meters
    - Firmware needs no aggressive WiFi power management, there is enough for AP mode and OTA firmware upgrade
  - ESP32-C3-MINI-1 module: 160MHz RISC-V core, 4MB Flash, 400kB RAM, 2.4GHz WiFi
  - 1mm² RGB LED
  - Two side-mounted buttons
    - Top: ESP32 reset
    - Bottom: AP mode, hold during reset to flash
  - Mounted by magnet on the IR port
  - Tested on an Austrian (Vbg) Kaifa MA309M
  - Designed in KiCad 7

I've had a small batch made, if anyone is interested in them, feel free to ping me. (david@beinder.at)

Prototype:
![front](https://user-images.githubusercontent.com/342955/224544583-15434392-1abf-40d9-b740-c316ce8bc80e.jpg)
![back](https://user-images.githubusercontent.com/342955/224544589-d196549e-8636-46ea-859a-723b9f368178.jpg)
![mount1](https://user-images.githubusercontent.com/342955/224550834-debf33a3-3fb9-4c1a-ac27-7f8ddc99f072.jpg)

![panel1](https://github.com/user-attachments/assets/de07476e-e120-447e-b0b0-897557dfce2f)

**Comment by Noschvie** [comment]
Hi David @dbeinder 
Would it be possible to provide a board like this but with a separated power supply ?
The M-Bus interface of netz-noe /EVN doesn‘t your configuration. Therefor an external power supply for the ESP32 is needed.
Thanks, Norbert

**Reply by dbeinder** [reply]
The [MBUS spec for KsmWest](https://www.vorarlbergnetz.at/media/KSMWest_H1_interface.pdf) (Vbg, Tirol, Sbg) also specifies max 6mA on the HAN port, but many meters actually supply more. I think that's just the lowest common denominator, so our grid operators have a wider selection when purchasing meters. My board simply takes advantage of the fact that Kaifa put in a pretty powerful HAN supply into their MA309 meter:
![image](https://user-images.githubusercontent.com/342955/224557670-4705fad3-9f5d-49d9-8647-1008a88df4fa.png)
If you have a MA309 too, I'm pretty sure my board will work for you, no matter what the EVN spec says ;)

If your meter really needs <6mA, the Pow-U device by the creators of this project is probably a far better solution for you.

My board can be powered by both USB and MBUS and simply uses the source with the higher voltage, however that is always going to be MBUS.
By desoldering one component, it could be modified to always use USB for power. I'll see if can change the layout so this can be done by cutting a trace on the PCB.

**Reply by GEAG** [reply]
I have recently been able to have a look at the document of the M-Bus standard spec EN 13757-2:2004 for the pysical layer
In Paragraph 4.3.1.1. "Max current (Imax)":

The current **Imax**, which the M-Bus master shall be able to provide, is defined as follows:

    N x single-unit-load + single-unit-short-circuit + single-unit-max-modulation
    
where
    N = max number of slaves (single-unit-loads) supported by the master on its bus
    single-unit-load = 1.5 mA
    single-unit-short-circuit = 100 mA
    single-unit-max-modulation = 20 mA

Any smart meter that claims in its specification that it complies with the M-Bus standard, should be able to provide this Imax amount of current. Otherwise the smart meter specs should let everybody know, that it does not fully comply with the standard and it should declare its restrictions.

For a smart meter with N=16, this formula delivers a value for Imax of 144 mA that seems to match the above graph from the Kaifa MA309 smart meter.

**Reply by ArnieO** [reply]
@GEAG It would be great if the meter manufacturers specified in more detail the power delivery capability of their HAN port. Unfortunately, that information is often missing or inadequate.

As an example, this is from the datasheet of the [M-bus HAN module for Kamstrup Omnipower meters](https://www.kamstrup.com/no-no/stroemloesninger/smartestroemmaalere/modules/han-nve):
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/987c0c00-ac93-41b3-8c5d-3ddcf5fbaee8)

**Comment by ArnieO** [comment]
Very impressive size - congratulations! 👏 👏 👏

**Comment by Noschvie** [comment]
@Soundrico what about this project?

**Reply by Soundrico** [reply]
wow looks also nice - wanna have

**Reply by Noschvie** [reply]
Suggest to use an ESP32 and  startup...

**Comment by Noschvie** [comment]
@comtechnik Are you familiar with this project? Use it for an EVN Kaifa smart meter.
